binder/views.py

```diff
@@ -1230,31 +1232,32 @@ def _store_m2m_field(self, obj, field, value):
 			old_ids = set(obj_field.values_list('id', flat=True))
 			new_ids = set(value)
 			for rmobj in obj_field.model.objects.filter(id__in=old_ids - new_ids):
+				rmobj_view = self.router.model_view(rmobj.__class__)()
 				method = getattr(rmobj, '_binder_unset_relation_{}'.format(obj_field.field.name), None)
 				if callable(method):
 					try:
-						method()
+						method(request)
 					except BinderValidationError as bve:
 						validation_errors.append(bve)
 				elif obj_field.field.null:
+					# A bit of a hack. We make a check, so we need to make sure that we do have the permissions
+					# for this
+					from binder.permissions.views import PermissionView
+					if isinstance(rmobj_view, PermissionView):
+						rmobj_view.scope_change(request, rmobj, {obj_field.field.name: None})
```

Comment:
Either way this is not a situation we haven't had before, because the actual logic isn't delegated to the subview. Personally, I think the best solution would be to not do the `setattr(` in this view, but use the `_store` method of the `rmobj_view` for instance. I however couldn't come up with a solution doing this which wouldn't have major side-effects.

The solution I was considering was to add the `scope_change` to the `ModelView` as well, which would solve the problem. I am not opposed do doing it this way. However, I felt like this solution  would contain the effects of this edge case much better. What is your opinion about this?